File: python3/Two Pointers/largest_container_brute_force.py
from typing import List


def largest_container_brute_force(heights: List[int]) -> int:
    n = len(heights)
    max_water = 0
    # Find the maximum amount of water stored between all pairs of
    # lines.
    for i in range(n):
        for j in range(i + 1, n):
            water = min(heights[i], heights[j]) * (j - i)
            max_water = max(max_water, water)
    return max_water

File: cpp/Two Pointers/largest_container_brute_force.cpp
#include <algorithm>
#include <vector>

int largestContainerBruteForce(std::vector<int>& heights) {
    int n = heights.size();
    int maxWater = 0;
    // Find the maximum amount of water stored between all pairs of 
    // lines. 
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int water = std::min(heights[i], heights[j]) * (j - i);
            maxWater = std::max(maxWater, water);
        }
    }
    return maxWater;
}